docs/docs/source-plugin-tutorial.md

```diff
@@ -0,0 +1,393 @@
+
+> **NOTE:** You should replace <YOUR_API_KEY_HERE> with your Pixabay API key.
+
+### Plugin behavior
+
+Your plugin will have the following behavior:
+
+* Accept config options like a Pixabay API key and a search query
+* Make an API request using the provided config options
+* Convert the data in the API response to Gatsby's node system
+
+### Setup a new Gatsby site
+
+Create a new Gatsby project and change directories into the new project you just created.
+
+```shell
+gatsby new source-tutorial-site
+cd source-tutorial-site
+```
+
+You're going to build your plugin as a "local" plugin that only exists for your project. Later on you'll learn how to publish a plugin to [npm](https://npmjs.com) so anyone can use it, but for now create a `plugins` directory and change into that directory:
+
+```shell
+mkdir plugins
+cd plugins
+```
+
+### Create a `plugins` folder
+
+The bare essentials of a plugin are a directory named after your plugin, which contains a `package.json` file and a `gatsby-node.js` file:
+
+```
+|-- plugins
+    |-- gatsby-source-pixabay
+        |-- gatsby-node.js
+        |-- package.json
+```
+
+Start by creating the directory and changing into it:
+
+```
+mkdir gatsby-source-pixabay
+cd gatsby-source-pixabay
+```
+
+### Create a `package.json` file
+
+Now create a `package.json` file, this describes your plugin and any third-party code it might depend on. `npm` has a command to create this file for you. Run:
+
+```shell
+npm init --yes
+```
+
+to create the file using default options.
+
+> **NOTE:** You can omit `--yes` if you'd like to specify the options yourself.
+
+### Add dependencies
+
+You'll use a couple of modules from npm to add some helper functionality. Install them with:
+
+```shell
+npm install node-fetch query-string --save
+```
+
+Open your `package.json` file and you'll see `node-fetch` and `query-string` have been added to a `dependencies` section at the end:
+
+```js
+  "dependencies": {
+    "node-fetch": "^2.1.2",
+    "query-string": "^6.0.0"
+  }
+```
+
+With the setup done, move on to adding the plugin's functionality.
+
+### Create a `gatsby-node.js` file
+
+Create a new file called `gatsby-node.js`, and add the following:
+
+```js
+const crypto = require("crypto");
+const fetch = require("node-fetch");
+const queryString = require("query-string");
+
+exports.sourceNodes = (
+  { boundActionCreators, createNodeId },
+  configOptions
+) => {
+  const { createNode } = boundActionCreators;
+
+  // Gatsby adds a configOption that's not needed for this plugin, delete it
+  delete configOptions.plugins;
+
+  // plugin code goes here...
+  console.log("Testing my plugin", configOptions);
+};
+```
+
+Let's break this down, you start by importing the dependencies that you added earlier (along with one built in dependency):
```

Comment:
*What did you just do by adding this code? You started by...

Reply:
👍 I dropped _just_ from your edit, so it's "What did you do by adding this code? You started by.."